### DISCUSSION TITLE
Should the density estimation be computed through KDE and not GMM ?

### DISCUSSION THREAD

**Discussion by ldesmet** [optuna/optuna#4452] [Q&A]
### Don't use GitHub Issues to ask support questions.

Hello everyone,

Quick question for you, TPE should used, as it is said in the name, Parzen Estimator to compute l and g density function ([here](https://docs.openvino.ai/2021.4/pot_compression_optimization_tpe_README.html#:~:text=Tree%2DStructured%20Parzen%20Estimator%20(TPE)%20algorithm%20is%20designed%20to,provide%20best%20possible%20latency%20improvement.))

> Two densities l(x1) and g(x2) are modeled using Parzen Estimators (also known as kernel density estimators) 

Instead, in the doc, you write 

> On each trial, for each parameter, TPE fits one Gaussian Mixture Model (GMM) l(x) to the set of parameter values associated with the best objective values, and another GMM g(x) to the remaining parameter values.

Do you compute l and g with GMM or KDE ?

Thanks a lot

**Comment by cross32768** [comment]
Thanks for your question. However, since issues are currently used for reporting bugs and feature requests,  could you ask questions in the [discussion](https://github.com/optuna/optuna/discussions)?  I'm sorry to trouble you.

**Reply by ldesmet** [reply]
Sorry could not have access to my git account yesterday, but I've seen someone did it :)
